{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "outputs": [],
   "source": [
    "%load_ext tensorboard"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:07:28.141668Z",
     "end_time": "2023-06-02T14:07:28.153247Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pip in /Users/mitya/venv/lib/python3.10/site-packages (23.1.2)\r\n",
      "Requirement already satisfied: setuptools in /Users/mitya/venv/lib/python3.10/site-packages (60.2.0)\r\n",
      "Collecting setuptools\r\n",
      "  Downloading setuptools-67.8.0-py3-none-any.whl (1.1 MB)\r\n",
      "\u001B[2K     \u001B[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001B[0m \u001B[32m1.1/1.1 MB\u001B[0m \u001B[31m4.7 MB/s\u001B[0m eta \u001B[36m0:00:00\u001B[0m00:01\u001B[0m00:01\u001B[0m\r\n",
      "\u001B[?25hRequirement already satisfied: wheel in /Users/mitya/venv/lib/python3.10/site-packages (0.37.1)\r\n",
      "Collecting wheel\r\n",
      "  Using cached wheel-0.40.0-py3-none-any.whl (64 kB)\r\n",
      "Installing collected packages: wheel, setuptools\r\n",
      "  Attempting uninstall: wheel\r\n",
      "    Found existing installation: wheel 0.37.1\r\n",
      "    Uninstalling wheel-0.37.1:\r\n",
      "      Successfully uninstalled wheel-0.37.1\r\n",
      "  Attempting uninstall: setuptools\r\n",
      "    Found existing installation: setuptools 60.2.0\r\n",
      "    Uninstalling setuptools-60.2.0:\r\n",
      "      Successfully uninstalled setuptools-60.2.0\r\n",
      "Successfully installed setuptools-67.8.0 wheel-0.40.0\r\n"
     ]
    }
   ],
   "source": [
    "!pip install --upgrade pip setuptools wheel"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:18:23.903420Z",
     "end_time": "2023-06-02T14:18:31.666004Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting opencv-python==3.4.13.47\r\n",
      "  Downloading opencv-python-3.4.13.47.tar.gz (87.6 MB)\r\n",
      "\u001B[2K     \u001B[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001B[0m \u001B[32m87.6/87.6 MB\u001B[0m \u001B[31m6.3 MB/s\u001B[0m eta \u001B[36m0:00:00\u001B[0m:00:01\u001B[0m00:01\u001B[0m\r\n",
      "\u001B[?25h  Installing build dependencies ... \u001B[?25lerror\r\n",
      "  \u001B[1;31merror\u001B[0m: \u001B[1msubprocess-exited-with-error\u001B[0m\r\n",
      "  \r\n",
      "  \u001B[31m×\u001B[0m \u001B[32mpip subprocess to install build dependencies\u001B[0m did not run successfully.\r\n",
      "  \u001B[31m│\u001B[0m exit code: \u001B[1;36m1\u001B[0m\r\n",
      "  \u001B[31m╰─>\u001B[0m \u001B[31m[236 lines of output]\u001B[0m\r\n",
      "  \u001B[31m   \u001B[0m Ignoring numpy: markers 'python_version == \"3.6\" and platform_machine != \"aarch64\"' don't match your environment\r\n",
      "  \u001B[31m   \u001B[0m Ignoring numpy: markers 'python_version >= \"3.6\" and sys_platform == \"linux\" and platform_machine == \"aarch64\"' don't match your environment\r\n",
      "  \u001B[31m   \u001B[0m Ignoring numpy: markers 'python_version == \"3.7\" and platform_machine != \"aarch64\"' don't match your environment\r\n",
      "  \u001B[31m   \u001B[0m Ignoring numpy: markers 'python_version == \"3.8\" and platform_machine != \"aarch64\"' don't match your environment\r\n",
      "  \u001B[31m   \u001B[0m Collecting setuptools\r\n",
      "  \u001B[31m   \u001B[0m   Using cached setuptools-67.8.0-py3-none-any.whl (1.1 MB)\r\n",
      "  \u001B[31m   \u001B[0m Collecting wheel\r\n",
      "  \u001B[31m   \u001B[0m   Using cached wheel-0.40.0-py3-none-any.whl (64 kB)\r\n",
      "  \u001B[31m   \u001B[0m Collecting scikit-build\r\n",
      "  \u001B[31m   \u001B[0m   Using cached scikit_build-0.17.6-py3-none-any.whl (84 kB)\r\n",
      "  \u001B[31m   \u001B[0m Collecting cmake\r\n",
      "  \u001B[31m   \u001B[0m   Using cached cmake-3.26.3-py2.py3-none-macosx_10_10_universal2.macosx_10_10_x86_64.macosx_11_0_arm64.macosx_11_0_universal2.whl (45.6 MB)\r\n",
      "  \u001B[31m   \u001B[0m Collecting pip\r\n",
      "  \u001B[31m   \u001B[0m   Using cached pip-23.1.2-py3-none-any.whl (2.1 MB)\r\n",
      "  \u001B[31m   \u001B[0m Collecting numpy==1.19.3\r\n",
      "  \u001B[31m   \u001B[0m   Downloading numpy-1.19.3.zip (7.3 MB)\r\n",
      "  \u001B[31m   \u001B[0m      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.3/7.3 MB 8.9 MB/s eta 0:00:00\r\n",
      "  \u001B[31m   \u001B[0m   Installing build dependencies: started\r\n",
      "  \u001B[31m   \u001B[0m   Installing build dependencies: finished with status 'done'\r\n",
      "  \u001B[31m   \u001B[0m   Getting requirements to build wheel: started\r\n",
      "  \u001B[31m   \u001B[0m   Getting requirements to build wheel: finished with status 'done'\r\n",
      "  \u001B[31m   \u001B[0m   Preparing metadata (pyproject.toml): started\r\n",
      "  \u001B[31m   \u001B[0m   Preparing metadata (pyproject.toml): finished with status 'error'\r\n",
      "  \u001B[31m   \u001B[0m   error: subprocess-exited-with-error\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m   × Preparing metadata (pyproject.toml) did not run successfully.\r\n",
      "  \u001B[31m   \u001B[0m   │ exit code: 1\r\n",
      "  \u001B[31m   \u001B[0m   ╰─> [198 lines of output]\r\n",
      "  \u001B[31m   \u001B[0m       setup.py:67: RuntimeWarning: NumPy 1.19.3 may not yet support Python 3.10.\r\n",
      "  \u001B[31m   \u001B[0m         warnings.warn(\r\n",
      "  \u001B[31m   \u001B[0m       Running from numpy source directory.\r\n",
      "  \u001B[31m   \u001B[0m       setup.py:480: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\r\n",
      "  \u001B[31m   \u001B[0m         run_build = parse_setuppy_commands()\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_bounded_integers.pxd.in\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_philox.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_bounded_integers.pyx.in\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_sfc64.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_mt19937.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/bit_generator.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/mtrand.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_generator.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_pcg64.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Processing numpy/random/_common.pyx\r\n",
      "  \u001B[31m   \u001B[0m       Cythonizing sources\r\n",
      "  \u001B[31m   \u001B[0m       blas_opt_info:\r\n",
      "  \u001B[31m   \u001B[0m       blas_mkl_info:\r\n",
      "  \u001B[31m   \u001B[0m       customize UnixCCompiler\r\n",
      "  \u001B[31m   \u001B[0m         libraries mkl_rt not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       blis_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries blis not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       openblas_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries openblas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_3_10_blas_threads_info:\r\n",
      "  \u001B[31m   \u001B[0m       Setting PTATLAS=ATLAS\r\n",
      "  \u001B[31m   \u001B[0m         libraries tatlas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_3_10_blas_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries satlas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_blas_threads_info:\r\n",
      "  \u001B[31m   \u001B[0m       Setting PTATLAS=ATLAS\r\n",
      "  \u001B[31m   \u001B[0m         libraries ptf77blas,ptcblas,atlas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_blas_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries f77blas,cblas,atlas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       accelerate_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries accelerate not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m       Library accelerate was not found. Ignoring\r\n",
      "  \u001B[31m   \u001B[0m         libraries veclib not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m       Library veclib was not found. Ignoring\r\n",
      "  \u001B[31m   \u001B[0m         FOUND:\r\n",
      "  \u001B[31m   \u001B[0m           extra_compile_args = ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers']\r\n",
      "  \u001B[31m   \u001B[0m           extra_link_args = ['-Wl,-framework', '-Wl,Accelerate']\r\n",
      "  \u001B[31m   \u001B[0m           define_macros = [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None)]\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m         FOUND:\r\n",
      "  \u001B[31m   \u001B[0m           extra_compile_args = ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers']\r\n",
      "  \u001B[31m   \u001B[0m           extra_link_args = ['-Wl,-framework', '-Wl,Accelerate']\r\n",
      "  \u001B[31m   \u001B[0m           define_macros = [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None)]\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       non-existing path in 'numpy/distutils': 'site.cfg'\r\n",
      "  \u001B[31m   \u001B[0m       lapack_opt_info:\r\n",
      "  \u001B[31m   \u001B[0m       lapack_mkl_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries mkl_rt not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       openblas_lapack_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries openblas not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       openblas_clapack_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries openblas,lapack not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       flame_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries flame not found in ['/Users/mitya/venv/lib', '/usr/local/lib', '/usr/lib']\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_3_10_threads_info:\r\n",
      "  \u001B[31m   \u001B[0m       Setting PTATLAS=ATLAS\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries tatlas,tatlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries tatlas,tatlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries tatlas,tatlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m       <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_3_10_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries satlas,satlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries satlas,satlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries satlas,satlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m       <class 'numpy.distutils.system_info.atlas_3_10_info'>\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_threads_info:\r\n",
      "  \u001B[31m   \u001B[0m       Setting PTATLAS=ATLAS\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries ptf77blas,ptcblas,atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries ptf77blas,ptcblas,atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m       <class 'numpy.distutils.system_info.atlas_threads_info'>\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       atlas_info:\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries f77blas,cblas,atlas not found in /Users/mitya/venv/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries f77blas,cblas,atlas not found in /usr/local/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries lapack_atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m         libraries f77blas,cblas,atlas not found in /usr/lib\r\n",
      "  \u001B[31m   \u001B[0m       <class 'numpy.distutils.system_info.atlas_info'>\r\n",
      "  \u001B[31m   \u001B[0m         NOT AVAILABLE\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m         FOUND:\r\n",
      "  \u001B[31m   \u001B[0m           extra_compile_args = ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers']\r\n",
      "  \u001B[31m   \u001B[0m           extra_link_args = ['-Wl,-framework', '-Wl,Accelerate']\r\n",
      "  \u001B[31m   \u001B[0m           define_macros = [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None)]\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m       /private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py:275: UserWarning: Unknown distribution option: 'define_macros'\r\n",
      "  \u001B[31m   \u001B[0m         warnings.warn(msg)\r\n",
      "  \u001B[31m   \u001B[0m       running dist_info\r\n",
      "  \u001B[31m   \u001B[0m       running build_src\r\n",
      "  \u001B[31m   \u001B[0m       build_src\r\n",
      "  \u001B[31m   \u001B[0m       building py_modules sources\r\n",
      "  \u001B[31m   \u001B[0m       creating build\r\n",
      "  \u001B[31m   \u001B[0m       creating build/src.macosx-10.9-universal2-3.10\r\n",
      "  \u001B[31m   \u001B[0m       creating build/src.macosx-10.9-universal2-3.10/numpy\r\n",
      "  \u001B[31m   \u001B[0m       creating build/src.macosx-10.9-universal2-3.10/numpy/distutils\r\n",
      "  \u001B[31m   \u001B[0m       building library \"npymath\" sources\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable gfortran\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable f95\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable f90\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable f77\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable xlf90\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable xlf\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable ifort\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable ifc\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable g77\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable g95\r\n",
      "  \u001B[31m   \u001B[0m       Could not locate executable pgfortran\r\n",
      "  \u001B[31m   \u001B[0m       don't know how to compile Fortran code on platform 'posix'\r\n",
      "  \u001B[31m   \u001B[0m       clang-14: error: the clang compiler does not support '-march=core2'\r\n",
      "  \u001B[31m   \u001B[0m       Traceback (most recent call last):\r\n",
      "  \u001B[31m   \u001B[0m         File \"/Users/mitya/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py\", line 353, in <module>\r\n",
      "  \u001B[31m   \u001B[0m           main()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/Users/mitya/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py\", line 335, in main\r\n",
      "  \u001B[31m   \u001B[0m           json_out['return_val'] = hook(**hook_input['kwargs'])\r\n",
      "  \u001B[31m   \u001B[0m         File \"/Users/mitya/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py\", line 149, in prepare_metadata_for_build_wheel\r\n",
      "  \u001B[31m   \u001B[0m           return hook(metadata_directory, config_settings)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py\", line 157, in prepare_metadata_for_build_wheel\r\n",
      "  \u001B[31m   \u001B[0m           self.run_setup()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py\", line 248, in run_setup\r\n",
      "  \u001B[31m   \u001B[0m           super(_BuildMetaLegacyBackend,\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/build_meta.py\", line 142, in run_setup\r\n",
      "  \u001B[31m   \u001B[0m           exec(compile(code, __file__, 'exec'), locals())\r\n",
      "  \u001B[31m   \u001B[0m         File \"setup.py\", line 508, in <module>\r\n",
      "  \u001B[31m   \u001B[0m           setup_package()\r\n",
      "  \u001B[31m   \u001B[0m         File \"setup.py\", line 500, in setup_package\r\n",
      "  \u001B[31m   \u001B[0m           setup(**metadata)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/core.py\", line 169, in setup\r\n",
      "  \u001B[31m   \u001B[0m           return old_setup(**new_attr)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/__init__.py\", line 165, in setup\r\n",
      "  \u001B[31m   \u001B[0m           return distutils.core.setup(**attrs)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py\", line 148, in setup\r\n",
      "  \u001B[31m   \u001B[0m           dist.run_commands()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py\", line 967, in run_commands\r\n",
      "  \u001B[31m   \u001B[0m           self.run_command(cmd)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py\", line 986, in run_command\r\n",
      "  \u001B[31m   \u001B[0m           cmd_obj.run()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/command/dist_info.py\", line 31, in run\r\n",
      "  \u001B[31m   \u001B[0m           egg_info.run()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/command/egg_info.py\", line 24, in run\r\n",
      "  \u001B[31m   \u001B[0m           self.run_command(\"build_src\")\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/cmd.py\", line 313, in run_command\r\n",
      "  \u001B[31m   \u001B[0m           self.distribution.run_command(command)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-build-env-m_vbpaa1/overlay/lib/python3.10/site-packages/setuptools/_distutils/dist.py\", line 986, in run_command\r\n",
      "  \u001B[31m   \u001B[0m           cmd_obj.run()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/command/build_src.py\", line 144, in run\r\n",
      "  \u001B[31m   \u001B[0m           self.build_sources()\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/command/build_src.py\", line 155, in build_sources\r\n",
      "  \u001B[31m   \u001B[0m           self.build_library_sources(*libname_info)\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/command/build_src.py\", line 288, in build_library_sources\r\n",
      "  \u001B[31m   \u001B[0m           sources = self.generate_sources(sources, (lib_name, build_info))\r\n",
      "  \u001B[31m   \u001B[0m         File \"/private/var/folders/6v/dfp997q13894cjhs_bzmt6j40000gn/T/pip-install-rgnfn4nx/numpy_1b66319f90cb4a03bfce495c73c22e66/numpy/distutils/command/build_src.py\", line 378, in generate_sources\r\n",
      "  \u001B[31m   \u001B[0m           source = func(extension, build_dir)\r\n",
      "  \u001B[31m   \u001B[0m         File \"numpy/core/setup.py\", line 663, in get_mathlib_info\r\n",
      "  \u001B[31m   \u001B[0m           raise RuntimeError(\"Broken toolchain: cannot link a simple C program\")\r\n",
      "  \u001B[31m   \u001B[0m       RuntimeError: Broken toolchain: cannot link a simple C program\r\n",
      "  \u001B[31m   \u001B[0m       [end of output]\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m   note: This error originates from a subprocess, and is likely not a problem with pip.\r\n",
      "  \u001B[31m   \u001B[0m error: metadata-generation-failed\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m × Encountered error while generating package metadata.\r\n",
      "  \u001B[31m   \u001B[0m ╰─> See above for output.\r\n",
      "  \u001B[31m   \u001B[0m \r\n",
      "  \u001B[31m   \u001B[0m note: This is an issue with the package mentioned above, not pip.\r\n",
      "  \u001B[31m   \u001B[0m hint: See above for details.\r\n",
      "  \u001B[31m   \u001B[0m \u001B[31m[end of output]\u001B[0m\r\n",
      "  \r\n",
      "  \u001B[1;35mnote\u001B[0m: This error originates from a subprocess, and is likely not a problem with pip.\r\n",
      "\u001B[1;31merror\u001B[0m: \u001B[1msubprocess-exited-with-error\u001B[0m\r\n",
      "\r\n",
      "\u001B[31m×\u001B[0m \u001B[32mpip subprocess to install build dependencies\u001B[0m did not run successfully.\r\n",
      "\u001B[31m│\u001B[0m exit code: \u001B[1;36m1\u001B[0m\r\n",
      "\u001B[31m╰─>\u001B[0m See above for output.\r\n",
      "\r\n",
      "\u001B[1;35mnote\u001B[0m: This error originates from a subprocess, and is likely not a problem with pip.\r\n",
      "\u001B[?25h"
     ]
    }
   ],
   "source": [
    "!pip install opencv-python==3.4.13.47"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:22:40.979291Z",
     "end_time": "2023-06-02T14:23:23.409660Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [],
   "source": [
    "import datetime\n",
    "import numpy as np\n",
    "import os\n",
    "import tensorflow as tf\n",
    "import matplotlib.pyplot as plt\n",
    "from tqdm import tqdm\n",
    "\n",
    "from tensorflow.keras import Model\n",
    "from tensorflow.keras.models import Sequential\n",
    "from tensorflow.keras.utils import to_categorical\n",
    "from tensorflow.keras.losses import categorical_crossentropy\n",
    "from tensorflow.keras.preprocessing.image import ImageDataGenerator\n",
    "from tensorflow.keras.layers import Dense, Flatten, Conv2D, MaxPooling2D, Dropout"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:09:26.331002Z",
     "end_time": "2023-06-02T14:09:26.365579Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [],
   "source": [
    "# import cv2\n",
    "import urllib\n",
    "import requests\n",
    "import PIL.Image\n",
    "import numpy as np\n",
    "from bs4 import BeautifulSoup"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:10:54.085426Z",
     "end_time": "2023-06-02T14:10:54.205907Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [],
   "source": [
    "#ship synset\n",
    "page = requests.get(\"http://www.image-net.org/api/text/imagenet.synset.geturls?wnid=n04194289\")\n",
    "soup = BeautifulSoup(page.content, 'html.parser')\n",
    "#bicycle synset\n",
    "bikes_page = requests.get(\"http://www.image-net.org/api/text/imagenet.synset.geturls?wnid=n02834778\")\n",
    "bikes_soup = BeautifulSoup(bikes_page.content, 'html.parser')\n",
    "\n",
    "str_soup=str(soup)\n",
    "split_urls=str_soup.split('\\r\\n')\n",
    "\n",
    "bikes_str_soup=str(bikes_soup)\n",
    "bikes_split_urls=bikes_str_soup.split('\\r\\n')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:11:59.272721Z",
     "end_time": "2023-06-02T14:12:01.461787Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [],
   "source": [
    "os.mkdir('./content')\n",
    "os.mkdir('./content/train')\n",
    "os.mkdir('./content/train/ships')\n",
    "os.mkdir('./content/train/bikes')\n",
    "os.mkdir('./content/validation')\n",
    "os.mkdir('./content/validation/ships')\n",
    "os.mkdir('./content/validation/bikes')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-06-02T14:11:10.637204Z",
     "end_time": "2023-06-02T14:11:10.689483Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "img_rows, img_cols = 32, 32\n",
    "input_shape = (img_rows, img_cols, 3)\n",
    "\n",
    "def url_to_image(url):\n",
    "    resp = urllib.request.urlopen(url)\n",
    "    image = np.asarray(bytearray(resp.read()), dtype=\"uint8\")\n",
    "    image = cv2.imdecode(image, cv2.IMREAD_COLOR)\n",
    "    return image\n",
    "\n",
    "n_of_training_images=100\n",
    "for progress in tqdm(range(n_of_training_images)):\n",
    "    if not split_urls[progress] == None:\n",
    "        try:\n",
    "            I = url_to_image(split_urls[progress])\n",
    "            if (len(I.shape))==3:\n",
    "                save_path = './content/train/ships/img'+str(progress)+'.jpg'\n",
    "                cv2.imwrite(save_path,I)\n",
    "        except:\n",
    "            None\n",
    "\n",
    "for progress in tqdm(range(n_of_training_images)):\n",
    "    if not bikes_split_urls[progress] == None:\n",
    "        try:\n",
    "            I = url_to_image(bikes_split_urls[progress])\n",
    "            if (len(I.shape))==3:\n",
    "                save_path = './content/train/bikes/img'+str(progress)+'.jpg'\n",
    "                cv2.imwrite(save_path,I)\n",
    "        except:\n",
    "            None\n",
    "\n",
    "\n",
    "for progress in tqdm(range(50)):\n",
    "    if not split_urls[progress] == None:\n",
    "        try:\n",
    "            I = url_to_image(split_urls[n_of_training_images+progress])\n",
    "            if (len(I.shape))==3:\n",
    "                save_path = './content/validation/ships/img'+str(progress)+'.jpg'\n",
    "                cv2.imwrite(save_path,I)\n",
    "        except:\n",
    "            None\n",
    "\n",
    "\n",
    "for progress in tqdm(range(50)):\n",
    "    if not bikes_split_urls[progress] == None:\n",
    "        try:\n",
    "            I = url_to_image(bikes_split_urls[n_of_training_images+progress])\n",
    "            if (len(I.shape))==3:\n",
    "                save_path = './content/validation/bikes/img'+str(progress)+'.jpg'\n",
    "                cv2.imwrite(save_path,I)\n",
    "        except:\n",
    "            None"
   ],
   "metadata": {
    "collapsed": false
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
